### ISSUE TITLE
Javascript still broken

### ISSUE THREAD

**Issue by jowr** [CoolProp/CoolProp#1754]
My new release 6.2,1 fixed the missing files issue for the Javascript builds, but I think the associated html page is broken. Is anyone able to debug: http://coolprop.sourceforge.net/jscript/ ?

**Comment by ibell** [comment]
The error I see is "Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://www.coolprop.sourceforge.net/jscript/coolprop.wasm. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing)" upon load of the page.  I saw this with developer tools in firefox

**Comment by jowr** [comment]
Any idea how to fix this? Or should we remove it from the next release?

**Comment by ibell** [comment]
One element of the problem is that the block in the header has

``` js
<script src="http://www.coolprop.sourceforge.net/jscript/coolprop.js"></script >
<!-- script src="coolprop.js"></script -->
```
when maybe we want

``` js
<!-- script src="http://www.coolprop.sourceforge.net/jscript/coolprop.js"></script -->
<script src="coolprop.js"></script>
```

But even so, copying the files locally and changing this code yields an error:

```wasm streaming compile failed: TypeError: Response has unsupported MIME type```

when I serve with:
```
python -m http.server 8000
```

**Comment by ibell** [comment]
And here is problem 2: the MIME type for .wasm files is not being served properly:

https://kripken.github.io/emscripten-site/docs/compiling/WebAssembly.html#web-server-setup

I think if we fix that we should be in business

**Comment by doomiris** [comment]
tested in NWJS, console as below

> coolprop.js:1 wasm streaming compile failed: TypeError: Failed to execute 'compile' on 'WebAssembly': Incorrect response MIME type. Expected 'application/wasm'.
> (anonymous) @ coolprop.js:1
> coolprop.js:1 falling back to ArrayBuffer instantiation
> (anonymous) @ coolprop.js:1
> wasm://wasm/00e90b2a:1295 Uncaught (in promise) RuntimeError: float unrepresentable in integer range
>     at wasm-function[1294]:111
>     at wasm-function[1274]:415
>     at wasm-function[1274]:177
>     at wasm-function[1274]:282
>     at wasm-function[1274]:177
>     at wasm-function[1274]:282
>     at wasm-function[3287]:15
>     at Object.Module.dynCall_iii (http://coolprop.sourceforge.net/jscript/coolprop.js:1:307319)
>     at invoke_iii (http://coolprop.sourceforge.net/jscript/coolprop.js:1:225301)
>     at wasm-function[1259]:1291

**Comment by ibell** [comment]
You'll want to give it a moment, but I tentatively think that I have identified a bug in the emscripten build of CoolProp, and also found a workaround (d12076f3c29ccdfc984861e74#diff-8fd02b4ce844f0684751774e160538c8R21).  We need to modify our server to serve the WASM properly (@jowr can you look into this?) but for local hosting, check this out: https://github.com/CoolProp/CoolProp/blob/master/Web/coolprop/wrappers/Javascript/index.rst

**Comment by ibell** [comment]
The problem is this: The sourceforge server doesn't serve the WASM with the right MIME type and CORS headers.  I fixed the WASM MIME type on the buildbot server by modifying the buildbot.tac file (http://www.coolprop.org/develop/buildbot.html#setting-mime-type-handler)

But I don't know how to fix the application of CORS headers to the response:
https://stackoverflow.com/questions/35254742/tornado-server-enable-cors-requests

Local hosting should still be fine.  It works for me.  Just grab the coolprop.js and coolprop.wasm files and drop them in your local directory and then launch your webserver.

**Comment by doomiris** [comment]
all output are 'Infinity' when i test local-hostingly. 
console shows:
```
Uncaught (in promise) RuntimeError: float unrepresentable in integer range
    at wasm-function[1296]:111
    at wasm-function[1276]:415
    at wasm-function[1276]:177
    at wasm-function[1276]:282
    at wasm-function[1276]:177
    at wasm-function[1276]:282
    at wasm-function[3293]:15
    at Object.Module.dynCall_iii (chrome-extension://engiigeekkjeioafcggelfekhkgcpibg/coolprop.js:1:307333)
    at invoke_iii (chrome-extension://engiigeekkjeioafcggelfekhkgcpibg/coolprop.js:1:225303)
    at wasm-function[1261]:1291
```

**Comment by ibell** [comment]
Can you give a bit more information about what you did?  I pulled the files from the dev web server with:
```
wget http://www.coolprop.dreamhosters.com:8010/binaries/Javascript/coolprop.js
wget http://www.coolprop.dreamhosters.com:8010/binaries/Javascript/coolprop.wasm
wget http://www.coolprop.dreamhosters.com:8010/binaries/Javascript/index.html
```
then set the path to the ``coolprop.js`` to be the local one in ``index.html``, and fired  up a little webserver with
``` python
import http.server
from http.server import BaseHTTPRequestHandler, HTTPServer

port=8000
print("Running on port %d" % port)

http.server.SimpleHTTPRequestHandler.extensions_map['.wasm'] = 'application/wasm'
httpd = HTTPServer(('localhost', port), http.server.SimpleHTTPRequestHandler)

print("Mapping \".wasm\" to \"{0}\"".format(http.server.SimpleHTTPRequestHandler.extensions_map['.wasm']))
httpd.serve_forever()
```
and ran with ``python serve.py``.  It runs perfectly with no problems in both firefox and chrome.  I still have no idea how to resolve the CORS issues.

**Comment by doomiris** [comment]
thanks for your reply. I intend to use coolProp in my Electron APP (tested with both Electron and NWJS). 
Now it seems OK after updating files above (dreamhosters),
 Previously, I got errors with 6.2.1version(download from sourceforge.net)

**Comment by ibell** [comment]
Indeed, I had to fix the JS file, so it is no surprise that the 6.2.1
version wouldn't work.  Good to hear it works for you now.

Could you perhaps make a very small demo of the use of Electron or NWJS
with CoolProp?  I've been curious to investigate this world and so far have
only dipped one or two toes in.

**Comment by doomiris** [comment]
i'm a beginner, but it's quite easy to get a 'hello world' :)   all you have to do is to prepare a package.json file.

1. download NWJS from https://nwjs.io/ (choose the package with SDK if you want to use console tool)
2. unzip, and create a new folder in the NWJS folder, name must be:  package.nw
3. put the above files(coolprop.js, coolprop.wasm, index.html) into the new folder
4. create package.json file, also put into package.nw, content as below
```
{
	"name": "cool prop test",
	"main": "index.html",
	"version": "1.2.3",
	"window": {
		"width": 800,
		"height": 600,
		"position": "center"
	}
}
```
5. run nw.exe,  (F12 key to open console)

setup Electron app is nearly the same.

**Comment by jowr** [comment]
Can this be included in the docs? I think we are close to a 6.2.2 release and it would be a very nice addition.

**Comment by doomiris** [comment]
sure, and i suggest that you should publish the javascript wrapper on NPM

**Comment by ibell** [comment]
That's a great idea, and I tried to figure out how to get CoolProp onto
cdnjs, but couldn't figure it out in the end.  Any tutorial?

**Comment by doomiris** [comment]
add to the repo(https://github.com/cdnjs/cdnjs) directly, but I never have done that.

**Comment by ibell** [comment]
Yeah, I couldn't figure out how to do it, so if you could figure it out and
do it, I'd welcome that.

**Comment by doomiris** [comment]
I've create pull request [https://github.com/cdnjs/cdnjs/pull/13318](https://github.com/cdnjs/cdnjs/pull/13318)

**Comment by ibell** [comment]
Cool, that's awesome!

**Comment by birdol** [comment]
wa o~~  it's great !

**Comment by jowr** [comment]
Thank you for your efforts. I have had a look at the PR you issued at cdnjs, but I think I do not understand. Where do the files come from? Did you compile them yourself? How would we update the library in the future?

**Comment by doomiris** [comment]
1. files were downloaded on Apr 1st from link here, and I assuming it as 6.2.2, I hope these don't bother you.
`wget http://www.coolprop.dreamhosters.com:8010/binaries/Javascript/coolprop.js
wget http://www.coolprop.dreamhosters.com:8010/binaries/Javascript/coolprop.wasm`
coolprop.min.js is generated with [web-minify-helper](https://github.com/PeterDaveHello/web-minify-helper)  
reason is that I've been failed ever with files download from [sourceforge.net](https://sourceforge.net/projects/coolprop/files/CoolProp/6.2.1/Javascript/), which is mentioned in above posts.
2.  you can update by submit PR, to Auto-Update, you had to publish files to NPM first, then add config in package.json, see [updating-an-existing-library](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#c-updating-an-existing-library)

**Comment by jowr** [comment]
OK, so updating should be done manually? I was wondering whether that was something that should be included in our release scripts.
I guess the best way to do it then is to 
 1) release CoolProp v6.3.0
 2) Update the PR at cdnjs manually with the files from sourceforge

Should we do that?

**Comment by doomiris** [comment]
yes, manually update.  and please consider publishing to NPM

**Comment by jowr** [comment]
This was fixed in v6.3.0 - @doomiris could you update the files for cdnjs with the official release binaries from SourceForge? 

https://sourceforge.net/projects/coolprop/files/CoolProp/6.3.0/Javascript/
http://coolprop.sourceforge.net/jscript/

**Comment by doomiris** [comment]
@jowr OK, i'll push update later, but the PR is still pending for review.
